### ISSUE TITLE
Identitical record name not imported properly

### ISSUE THREAD

**Issue by nilius** [julianpeeters/avrohugger#114]
This is probably a duplicate to https://github.com/julianpeeters/avrohugger/issues/69.

I am given a topic with a schema which contains the class name "Order" in two positions:
```{
  "type": "record",
  "name": "Order",
  "namespace": "com.some",
  "fields": [
    {
      "name": "payload",
      "type": {
        "type": "record",
        "name": "Order",
        "namespace": "com.other",
        "fields": [...]
```
The generated class uses an import which is shaded by the class defined in the file:

```
/** MACHINE-GENERATED FROM AVRO SCHEMA. DO NOT EDIT DIRECTLY */
package com.some

import scala.annotation.switch
import com.other.{Order}
case class Order(var id: String, var timestamp: Long, var source: String, var `type`: String, var payload: Order) extends [...]
```
Here, `payload: Order` references the wrong class Order, it should reference the com.other.Order from the import but that import is shaded by the class com.some.Order from this file.

A fix, as suggested in https://github.com/julianpeeters/avrohugger/issues/69, would be not to use imports but to to generate 
```case class Order(var id: String, var timestamp: Long, var source: String, var `type`: String, var payload: com.other.Order) extends [...]```

Any comments/ídeas, especially since issue 69 is from June 2017?

**Comment by julianpeeters** [comment]
awesome, thanks for the report and the bump @nilius . I don't personally have a need for this so it's been on my back-burner, but I think I'll have some time this weekend to take a look, we'll see how far I get.

**Comment by nilius** [comment]
Hi Julian, maybe this weekend?  : )

**Comment by julianpeeters** [comment]
Sorry @nilius , this is still pretty far down on my queue. I don't have a good estimate of a timeframe. PRs welcome, of course.

**Comment by reidrac** [comment]
Hi @julianpeeters, we have been stung by this one pretty hard. Any update you can share about this? Thanks!

**Comment by makubi** [comment]
I've set up a test at https://github.com/makubi/avrohugger-test that is using the avrohugger-core library

**Comment by ismailBenammar** [comment]
Hi @julianpeeters, i have come to add this pull request https://github.com/julianpeeters/avrohugger/pull/169 in order to fix the problem. I hope that won't Contrary your policy. 
Happy review :-)

**Comment by julianpeeters** [comment]
Thanks, all. Fix is now available in avrohugger 1.4.0 and sbt-avrohugger-2.4.0
